tests/misc_env/install_vault.py

```diff
@@ -138,63 +141,83 @@ def _write_remote_file(node: CephNode, file_name: str, content: str) -> None:
     """
     Copies the provided content to the specified file on the given node.
     """
-    LOG.debug(f"{node.shortname}: Writing to remote file {file_name}")
+    LOG.info(f"{node.shortname}: Writing to remote file {file_name}")
     file_handle = node.remote_file(sudo=True, file_mode="w", file_name=file_name)
     file_handle.write(data=content)
     file_handle.flush()
     file_handle.close()
 
 
-def _install_agent(cluster: Ceph, config: Dict) -> None:
+def _install_agent(cluster: Ceph, config: Dict, *, cloud_type: str) -> None:
     """
     Installs and configures the vault-agent on all RGW nodes.
     """
     rgw_nodes = cluster.get_nodes(role="rgw")
     for node in rgw_nodes:
-        LOG.debug(f"{node.shortname}: Installing and configuring Vault agent")
-        _install_vault_packages(node)
+        LOG.info(
+            f"{node.shortname}: Installing and configuring Vault agent (cloud_type={cloud_type})"
+        )
+        _install_vault_packages(node, cloud_type=cloud_type)
         _create_agent_config(node, config)
         _create_agent_systemd(node)
 
 
-def _install_vault_packages(node: CephNode) -> None:
+def _install_vault_packages(node: CephNode, *, cloud_type: str) -> None:
     """
-    Installs the required packages for Vault based on IBM-Ceph or RH-Ceph environment.
+    Installs the required packages for Vault based on the explicit cloud_type
+    ('ibmc' or 'openstack') selection from cephci.yaml instead of heuristics.
     """
     try:
-        out = node.exec_command(sudo=True, cmd="podman ps", check_ec=False)[1]
-
-        if "ibm-ceph" in out:
-            LOG.debug(f"{node.shortname}: Detected IBM-Ceph environment")
-
-            repo_cmd = """
-cat <<EOF | sudo tee /etc/yum.repos.d/hashicorp.repo
+        if cloud_type == "ibmc":
+            LOG.info(
+                f"{node.shortname}: Using HashiCorp official repo for IBM-Ceph (cloud_type=ibmc)"
+            )
+            repo_cmd = r"""
```

Comment:
better to pass this repo file as `custom-config-file`

Reply:
sure @vamahaja , I will take that in the next PR, this is to resolve the TFA issue